mmlearn/datasets/core/dataset_info.py

```diff
@@ -0,0 +1,116 @@
+"""Saving the necessary information about a dataset for evaluation tasks."""
+
+from typing import Any, Dict, Optional
+
+
+class DatasetInfo:
```

Comment:
Instead of this class, how about defining `num_classes` and `label2id` in each dataset class (probably as a class variable), then have `CombinedDataset` merge it if it's available?

Reply:
I think this will put more work for user in defining each dataset. Right now the user only needs to define one dictionary that is for label_mapping and everything else such as number of classes and the label embeddings of that dataset will be handled with DatasetInfo